### PR TITLE
[providers] Refactor code to reuse common things

### PR DESCRIPTION
**Pull request by n1chre**
We put common stuff inside providers/lib to be shared between all providers.
Notable changes:
- refactored snyk2nvd a bit. Created providers/snyk/api package to do the token handling and fetching from their api. To be consistent with other providers and to use the dependancy on providers/lib/download
- refactored *2nvd binaries so they all use the same convention to create clients and use the same flags
- refactored idefense client so it behaves in the same way as all others
- mv cmd/nvdsync/datafeed to providers/nvd; also uses download.Client()

Test: go build && go test && ran all *2nvd manually, all works

### CODE REVIEW

**Comment by skogtwin** [comment]
Frankly, I'm not sure if such consistency worth adding 67 lines of code and another level of indirection...

**Comment by n1chre** [comment]
@skogtwin the only real "added lines" are the new package file and this:
```
-	resp, err := http.DefaultClient.Do(req)
+	httpClient, err := common.HttpClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "can't obtain http client")
+	}
+	resp, err := httpClient.Do(req)
```
imho 67 lines are nothing compared to the gains we get by having consistent codebase and an easy way to use a different client internally for all converters. 

Also: will change the package name to smth else. And add license headers, forgot them again

**Comment by n1chre** [comment]
@skogtwin went from 67 to -44. Hopefully it's ok now :D

File: providers/common/http.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package common
```
[the remark this answers is not in the document]
Comment (n1chre) [reply providers/common/http.go]: outdated, it's named download now

File: cmd/snyk2nvd/snyk2nvd.go [hunk 2]
```diff
@@ -86,6 +83,12 @@ func obtainFeed(download bool) (io.ReadCloser, error) {
 		if token == "" {
 			log.Fatal("Please set SNYK_TOKEN in environment")
 		}
+		parts := strings.SplitN(strings.TrimSuffix(token, "\n"), ":", 2)
```
[the remark this answers is not in the document]
Comment (n1chre) [reply cmd/snyk2nvd/snyk2nvd.go]: will do

**Pull request merged**